pinot-plugins/pinot-input-format/pinot-protobuf/src/main/java/org/apache/pinot/plugin/inputformat/protobuf/ProtoBufUtils.java

```diff
@@ -65,4 +75,114 @@ public static File createLocalFile(URI srcURI, File dstDir) {
         + "buffer descriptor {}", dstFile.getAbsolutePath(), srcURI);
     return dstFile;
   }
+
+  public static Descriptors.Descriptor buildDescriptor(String messageName, String descFilePathOpt) {
+    if (descFilePathOpt != null && !descFilePathOpt.isEmpty()) {
+      return getDescriptor(descFilePathOpt, messageName);
+    } else {
+      return buildDescriptorFromJavaClass(messageName);
+    }
+  }
+
+  /**
+   *  Loads the given protobuf class and returns Protobuf descriptor for it.
+   */
+  private static Descriptors.Descriptor buildDescriptorFromJavaClass(String protobufClassName) {
+    Class<?> protobufClass;
+    try {
+      protobufClass = Class.forName(protobufClassName);
```

Comment:
How is the user expected to upload class? e.g. if 30 pinot servers are running, users should copy their jar to all of them and then restart?

Reply:
@KKcorps yes, Isn't it the same with the descriptorFile option? If the import is done via Spark or Hadoop, users can pass the jar part of the —jars option. 

This change eliminates the need to create a .desc file. The user would have a .proto jar rather than a .desc file. If you believe the descriptorFile-based option is useful, we can continue to provide users with both options.

Reply:
Using third-party jars is much more risky. Plus it would require a server restart. Descriptor files don't require server restart.

Reply:
@KKcorps That makes sense; I hadn't considered the possibility of a server restart. will close it.